test/python/synthesis/test_synthesis.py

```diff
@@ -1460,6 +1462,35 @@ def __init__(self, theta, label=None):
         circ = decomposer(unitary)
         self.assertEqual(Operator(unitary), Operator(circ))
 
+    def test_unitary_custom_gate_raises(self):
+        """Test that a custom gate raises an exception, as it's not equivalent to an RXX gate"""
+
+        class CustomXYGate(Gate):
+            """Custom Gate subclass that's not a standard gate and not RXX equivalent"""
+
+            _standard_gate = None
+
+            def __init__(self, theta: ParameterValueType, label=None):
+                """Create new custom rotstion XY gate."""
+                super().__init__("MyCustomXYGate", 2, [theta])
+
+            def __array__(self, dtype=None):
+                """Return a Numpy.array for the custom gate."""
+                theta = self.params[0]
+                cos = math.cos(theta)
+                isin = 1j * math.sin(theta)
+                return np.array(
+                    [[1, 0, 0, 0], [0, cos, -isin, 0], [0, -isin, cos, 0], [0, 0, 0, 1]],
+                    dtype=dtype,
+                )
+
+            def inverse(self, annotated: bool = False):
+                return CustomXYGate(-self.params[0])
+
+        with self.assertRaises(QiskitError) as exc:
+            _ = TwoQubitControlledUDecomposer(CustomXYGate)
+        self.assertIn("Specialization: ControlledEquiv calculated fidelity", exc.exception.message)
```

Comment:
FWIW, you can do this with `self.assertRaisesRegex` https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex by specifying a regular expression that matches the target error message. This would be something like:

```suggestion
        with self.assertRaisesRegex(QiskitError, "ControlledEquiv calculated fidelity"):
            TwoQubitControlledUDecomposer(CustomXYGate)
```

but you can get more sophisticated with the regular expression to atch more exactly on the error message if you want.

Reply:
done in [8949f7f](https://github.com/Qiskit/qiskit/pull/13139/commits/8949f7f4c3a0c02318f368f82eea3d901aa5ac26)